### PR TITLE
Separate RobotBase from MJBotsInterface

### PR DESCRIPTION
**Pull request by jdcaporale**
(Pull Request 2 of 3 in a sequence: builds from PR #30  and originally in PR #22 )

Adds the RobotInterface Class. Its role is to separate MJBots specfic communication and hardware interfacing from  robot state for control. The MJBots hardware interface shares ownership of the JointBase/Moteus pointers and uses them for communications and the robot is free to set trques and to be derived to include any future features (limbs, state estimators, etc.).
Also, adds a simple derived robot example. 

TODO:
- [x] Upgrade the example to match our design intention comments (e.g. [this](https://github.com/KodlabPenn/kodlab_mjbots_sdk/pull/22#discussion_r931535027) or [this](https://github.com/KodlabPenn/kodlab_mjbots_sdk/pull/22#discussion_r926131102) on previous PR

### CODE REVIEW

File: include/examples/simple_robot.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+{
+
+    using kodlab::RobotInterface::RobotInterface;
+
+public:
+    int mode = 0; // Member variables encode whatever added state we need
+    // Set up robot update function for state and torques
+    void Update() override
+    {
+        std::vector<float> torques(num_joints_, 0);
```
Comment (ShaneRozenLevy) [inline include/examples/simple_robot.h:26]: I think this is fine, but we really want to make the SimpleRobot demonstrate more about how we expect the `RobotInterface` to be used.

File: include/kodlab_mjbots_sdk/mjbots_robot_interface.h [hunk 2]
```diff
@@ -31,25 +31,23 @@ struct RealtimeParams {
   int lcm_cpu = 0;
 };
 
-class MjbotsRobotInterface {
+/*!
+ * @brief A class that allows interaction with the Mjbots Moteus Motor Controllers
+ * 
+ */
+class MjbotsRobotInterface  {
```
Comment (ShaneRozenLevy) [inline include/kodlab_mjbots_sdk/mjbots_robot_interface.h:38]: Lets rename to mjbots hardware interface.

File: include/kodlab_mjbots_sdk/robot_interface.h [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+#include "kodlab_mjbots_sdk/joint_base.h"
+#include "kodlab_mjbots_sdk/soft_start.h"
+
+namespace kodlab
+{
+    /*!
+     * @brief Sets up a kodlab robot interface class 
+     * 
+     */
+    class RobotInterface {
```
Comment (ShaneRozenLevy) [inline include/kodlab_mjbots_sdk/robot_interface.h:25]: Lets name something like RobotBase.

File: include/kodlab_mjbots_sdk/robot_interface.h [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    public:
+        static const int KILL_ROBOT = -1;
```
Comment (ShaneRozenLevy) [inline include/kodlab_mjbots_sdk/robot_interface.h:27]: Add a comment explaining what this is/does.

**Review by ShaneRozenLevy — CHANGES_REQUESTED** [review]
Need to update the readme. Otherwise besides some naming issues it looks good.

File: include/examples/simple_robot.h (hunk 1 above)
Comment (ethanmusser) [reply include/examples/simple_robot.h:26]: Agreed, we should do a formatting run (see #35).

File: include/kodlab_mjbots_sdk/mjbots_robot_interface.h (hunk 2 above)
Comment (ethanmusser) [reply include/kodlab_mjbots_sdk/mjbots_robot_interface.h:38]: This is covered in part 3/3 of the PR

File: include/kodlab_mjbots_sdk/robot_interface.h (hunk 3 above)
Comment (ethanmusser) [reply include/kodlab_mjbots_sdk/robot_interface.h:25]: This is covered in part 3/3 of the PR

File: include/examples/simple_robot.h (hunk 1 above)
Comment (jdcaporale) [reply include/examples/simple_robot.h:26]: Kept in issue https://github.com/KodlabPenn/kodlab_mjbots_sdk/issues/36#issue-1326662355 
Will make a PR just for this change.

File: include/kodlab_mjbots_sdk/robot_interface.h (hunk 4 above)
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/robot_interface.h:27]:
```suggestion
        static const int KILL_ROBOT = -1;  // Kill mode/behavior index, which will be used to signal E-Stopping the robot
```

Comment (ethanmusser) [reply include/kodlab_mjbots_sdk/robot_interface.h]: Resolved in 571f61791965a974d1ae6976b4ec54d24ca269b0.

Comment (ethanmusser) [reply include/kodlab_mjbots_sdk/robot_interface.h]: Resolved in 571f61791965a974d1ae6976b4ec54d24ca269b0.

**Comment by ethanmusser** [comment]
README.md updated in 4b9c6b356d60f8d9a524c07af70b76d5b2ddaa11.

**Review by ethanmusser — APPROVED** [review]
All problems have been remedied or addressed in seperate issues.  LGTM.

File: README.md [hunk 5]
```diff
@@ -9,26 +9,36 @@ for better realtime performance
...
+
 ## MjbotsControlLoop:
-To use the `MjbotsControlLoop` create a class which inherits the `MjbotsControlLoop`
-and implements `CalcTorques` to set the torques in the robot object. 
+To use the Mjbots control loop, create a class which inherits the 
+`MjbotsControlLoop` object and implements `CalcTorques` to set the torques in 
+the robot object as follows.
+
+```cpp
+class SimpleRobotControlLoop : public kodlab::mjbots::MjbotsControlLoop
```
Comment (jdcaporale) [inline README.md:29]: SimpleRobot is a derived class and should either be in the template or this default version should just be RobotControlLoop.

**Pull request merged**